temporalio/lib/temporalio/client.rb

```diff
@@ -0,0 +1,282 @@
+  # Fiber-compatible.
+  class Client
+    # Options as returned from {dup_options} for +**to_h+ splat use in {initialize}. See {initialize} for details.
+    Options = Struct.new(
+      :connection,
+      :namespace,
+      :data_converter,
+      :interceptors,
+      :default_workflow_query_reject_condition,
+      keyword_init: true
+    )
+
+    # Connect to Temporal server. This is a shortcut for +Connection.new+ followed by +Client.new+.
+    #
+    # @param target_host [String] +host:port+ for the Temporal server. For local development, this is often
+    #   +localhost:7233+.
+    # @param namespace [String] Namespace to use for client calls.
+    # @param api_key [String, nil] API key for Temporal. This becomes the +Authorization+ HTTP header with +"Bearer "+
+    #   prepended. This is only set if RPC metadata doesn't already have an +authorization+ key.
+    # @param tls [Boolean, Connection::TLSOptions] If false, do not use TLS. If true, use system default TLS options. If
+    #   TLS options are present, those TLS options will be used.
+    # @param data_converter [Converters::DataConverter] Data converter to use for all data conversions to/from payloads.
+    # @param interceptors [Array<Interceptor>] Set of interceptors that are chained together to allow intercepting of
+    #   client calls. The earlier interceptors wrap the later ones. Any interceptors that also implement
+    #   {Worker::Interceptor} will be used as worker interceptors too so they should not be given separately when
+    #   creating a worker.
+    # @param default_workflow_query_reject_condition [Api::Enums::V1::QueryRejectCondition, nil] Default rejection
+    #   condition for workflow queries if not set during query. See {WorkflowHandle.query} for details on the
+    #   rejection condition.
+    # @param rpc_metadata [Hash{String=>String}] Headers to use for all calls to the server. Keys here can be overriden
+    #   by per-call RPC metadata keys.
+    # @param rpc_retry [Connection::RPCRetryOptions] Retry options for direct service calls (when opted in) or all
+    #   high-level calls made by this client (which all opt-in to retries by default).
+    # @param identity [String] Identity for this client.
+    # @param keep_alive [Connection::KeepAliveOptions] Keep-alive options for the client connection. Can be set to +nil+
+    #   to disable.
+    # @param http_connect_proxy [Connection::HTTPConnectProxyOptions] Options for HTTP CONNECT proxy.
+    # @param runtime [Runtime] Runtime for this client.
+    # @param lazy_connect [Boolean] If true, the client will not connect until the first call is attempted or a worker
+    #   is created with it. Lazy clients cannot be used for workers if they have not performed a connection.
+    #
+    # @return [Client] Connected client.
+    #
+    # @see Connection.initialize
+    # @see initialize
+    def self.connect(
+      target_host,
+      namespace,
```

Comment:
Note how unlike other SDKs, we do not default this to `default`. There are many cloud and/or multi-namespace self-hosted users these days and so the `default` namespace, like the `localhost` endpoint default from before, makes less sense.